src/ra_machine.erl

```diff
@@ -135,6 +135,8 @@
     {log, [ra_index()], fun(([user_command()]) -> effects())} |
     {log, [ra_index()], fun(([user_command()]) -> effects()), {local, node()}} |
     {release_cursor, ra_index(), state()} |
+    {release_cursor, ra_index()} |
```

Comment:
I think these two new effects need to be added to `ra_server:filter_follower_effects/1` else it may be that followers won't be able to emit these.

`filter_follower_effects/1` is an old function and arguably it would be better and more intuitive to do the filtering of which effects should and shouldn't be executed based on the Raft state in `ra_server_proc:handle_effects/`

Reply:
Might be nice to get some kind of full integration test in for checkpoints, just to validate the happy path. Perhaps in `coordionation_SUITE`